docs/dev_contribution.rst

```diff
@@ -55,89 +57,85 @@ General labels currently used for libCellML are:
    :align: center
    :alt: Git branching illustration.
 
-   `Git <https://git-scm.com/>`_ branching illustration.
+   Git branching illustration.
 
-Following this process will make it easier to have multiple topic branches at once and keep them in-sync with the :term:`prime repository` ``develop`` branch, which will in turn make it easier to manage multiple pull requests.
+Following this process will make it easier to have multiple topic branches at once and keep them in-sync with the :term:`prime repository` :code:`develop` branch, which will in turn make it easier to manage multiple pull requests.
 
-The following `Git <https://git-scm.com/>`_ command line commands show an example of how to create a topic branch for fixing a (hypothetical) bug described in `issue #123`::
+The following Git commands show an example of how to create a topic branch for fixing a (hypothetical) bug described in `issue #123`::
 
   git fetch prime develop
   git checkout develop # Not required if already on develop branch
   git merge prime/develop
   git checkout -b issue123
 
-Test Driven Development
-=======================
-
-Test driven development entails writing a test that covers the intended functionality (this may require a suite of tests to be written) and no more.
-The tests will require some skeleton implementation so that the test(s) can compile but by definition not pass, at least not pass all the tests.
-The purpose of this is two-fold:
-
-1. write the test(s) first, set out the intended design that can be shared through a pull request; and
-#. implement the skeleton that will include the documentation clearly describing the intended purpose.
 
-Following this contribution process allows others to comment and make corrections before time is spent on the functional code.
+Test-driven development
+=======================
+Test-driven development entails writing a test (or suite of tests) that covers the intended functionality, and no more.
+The tests will require some skeleton implementation so that they can compile, but in the first instance they probably won't pass.
+Writing the test(s) first shows the other developers the new behaviour that you intend to implement.
+By making a pull request containing only these tests, other developers can then comment and discuss until everyone is happy with the new behaviour.
+Following this contribution process allows for corrections before time is spent on the functional code.
 
 It may be necessary to refactor the current design to enable the easiest possible way to add the new feature.
 This is a good thing as the quality of the design improves and this makes it easier to work with in the future.
 Refactoring means improving the code without adding features, and the tests provide validation that the refactored code performs as well as before.
 
 For simple or obvious bugs, which have fallen through the testing gaps, just the implementation is fine.
+**TODO** KRM I don't understand this sentence?
```

Comment:
It says for simple or (now) obvious bugs you don't need to write a test just fix the implementation.

Reply:
Doesn't that go against all the testing rationale?  I thought that was kinda the point - that the tests could be relied upon to test whatever bugs we knew about?